smoothie.js

```diff
@@ -671,6 +671,11 @@
       var timeSeries = this.seriesSet[d].timeSeries,
           dataSet = timeSeries.data,
           seriesOptions = this.seriesSet[d].options;
+          
+      if (seriesOptions.lineWidth === 0) {
+        context.restore();
+        continue;
```

Comment:
If you continue here, then the call to `dropOldData` call below is skipped too, and the data will build up indefinitely in the time series object.